src/main/java/io/confluent/connect/elasticsearch/Mapping.java

```diff
@@ -27,94 +23,206 @@
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.DOUBLE_TYPE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.FLOAT_TYPE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.INTEGER_TYPE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.KEYWORD_TYPE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.LONG_TYPE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.MAP_KEY;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.MAP_VALUE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.SHORT_TYPE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.STRING_TYPE;
-import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConstants.TEXT_TYPE;
 
 public class Mapping {
 
+  // Elasticsearch types
+  public static final String BOOLEAN_TYPE = "boolean";
+  public static final String BYTE_TYPE = "byte";
+  public static final String BINARY_TYPE = "binary";
+  public static final String SHORT_TYPE = "short";
+  public static final String INTEGER_TYPE = "integer";
+  public static final String LONG_TYPE = "long";
+  public static final String FLOAT_TYPE = "float";
+  public static final String DOUBLE_TYPE = "double";
+  public static final String STRING_TYPE = "string";
+  public static final String TEXT_TYPE = "text";
+  public static final String KEYWORD_TYPE = "keyword";
+  public static final String DATE_TYPE = "date";
+
+  // Elasticsearch mapping fields
+  private static final String DEFAULT_VALUE_FIELD = "null_value";
+  private static final String FIELDS_FIELD = "fields";
+  private static final String IGNORE_ABOVE_FIELD = "ignore_above";
+  public static final String KEY_FIELD = "key";
+  private static final String KEYWORD_FIELD = "keyword";
+  private static final String PROPERTIES_FIELD = "properties";
+  private static final String TYPE_FIELD = "type";
+  public static final String VALUE_FIELD = "value";
+
   /**
-   * Create an explicit mapping.
+   * Build mapping from the provided schema.
    *
-   * @param client The client to connect to Elasticsearch.
-   * @param index  The index to write to Elasticsearch.
-   * @param type   The type to create mapping for.
-   * @param schema The schema used to infer mapping.
-   * @throws IOException from underlying JestClient
+   * @param schema The schema used to build the mapping.
    */
-  public static void createMapping(
-      ElasticsearchClient client,
-      String index,
-      String type,
-      Schema schema
-  )
-      throws IOException {
-    client.createMapping(index, type, schema);
+  public static XContentBuilder buildMapping(Schema schema) {
+    try {
+      XContentBuilder builder = XContentFactory.jsonBuilder();
+      builder.startObject();
+      {
```

Comment:
Why do we need these curly brackets?

Reply:
this is the convention that ES documentation used, so I kept for
1. consistency
2. readability (helps visualize nested objects)